src/async_fuse/memfs/kv_engine.rs

```diff
@@ -180,23 +248,229 @@ pub trait MetaTxn {
+/// Currently support 'lease' and `prev_kv`
+/// `lease` is used to set the lease of the key
+/// `prev_kv` is used to return the previous key-value pair
+#[allow(dead_code)]
+#[derive(Debug, Eq, PartialEq)]
+pub struct SetOption {
+    /// The lease of the key
+    pub(crate) lease: Option<i64>,
+    /// Whether to return the previous key-value pair
+    pub(crate) prev_kv: bool,
+}
+
+impl SetOption {
+    #[allow(dead_code)]
+    #[must_use]
+    /// Create a new `SetOption`
+    /// Default lease is None, `prev_kv` is false
+    fn new() -> Self {
+        Self {
+            lease: None,
+            prev_kv: false,
+        }
+    }
+
+    #[allow(dead_code)]
+    #[must_use]
+    /// Set the lease of the key
+    fn with_lease(mut self, lease: i64) -> Self {
+        self.lease = Some(lease);
+        self
+    }
+
+    #[allow(dead_code)]
+    #[must_use]
+    /// Set whether to return the previous key-value pair
+    fn with_prev_kv(mut self) -> Self {
+        self.prev_kv = true;
+        self
+    }
+}
+
+/// The option of 'delete' operation
+/// Currently support `prev_kv` and `range_end`
+/// `prev_kv` is used to return the previous key-value pair
+/// `range_end` is used to delete all keys in the range [key, `range_end`)
+#[allow(dead_code)]
+#[derive(Debug, Eq, PartialEq)]
+pub struct DeleteOption {
+    /// Whether to return the previous key-value pair
+    pub(crate) prev_kv: bool,
+    /// The range end of the delete operation
+    pub(crate) range_end: Option<Vec<u8>>,
+}
+
+impl DeleteOption {
+    #[allow(dead_code)]
+    #[must_use]
+    /// Create a new `DeleteOption`
+    fn new() -> Self {
+        Self {
+            prev_kv: false,
+            range_end: None,
+        }
+    }
+
+    #[allow(dead_code)]
+    #[must_use]
+    /// Set whether to return the previous key-value pair
+    fn with_prev_kv(mut self) -> Self {
+        self.prev_kv = true;
+        self
+    }
+
+    #[allow(dead_code)]
+    #[must_use]
+    /// Set the range end of the delete operation
+    fn with_range_end(mut self, range_end: Vec<u8>) -> Self {
+        self.range_end = Some(range_end);
+        self
+    }
+
+    /// Delete all keys
+    #[allow(dead_code)]
+    #[must_use]
+    /// Set the range end of the delete operation
+    fn with_all_keys(mut self) -> Self {
+        self.range_end = Some(vec![0xff]);
+        self
+    }
+}
+
+/// `KeyRange` is an abstraction for describing etcd key of various types.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Debug)]
+#[allow(dead_code)]
+pub struct KeyRange {
+    /// The first key of the range and should be non-empty
+    pub(crate) key: Vec<u8>,
+    /// The key following the last key of the range
+    pub(crate) range_end: Vec<u8>,
+}
+
+impl KeyRange {
+    /// Creates a new `KeyRange` for describing a range of multiple keys.
+    #[inline]
+    #[allow(dead_code)]
+    pub fn range<K, R>(key: K, range_end: R) -> Self
+    where
+        K: Into<Vec<u8>>,
+        R: Into<Vec<u8>>,
```

Comment:
The `Into` trait usually cause a memory copy. Can we use `AsRef` here as we usually don't consume the `KeyRange` parameter.